conftest.py

```diff
@@ -760,3 +780,171 @@ def cifs_iso_sr(host, cifs_iso_device_config):
     yield sr
     # teardown
     sr.forget()
+
+
+# Session-scoped fixture to create a shared log directory for all artifacts
+@pytest.fixture(scope='session')
+def session_log_dir():
+    """
+    Create and return a session-wide log directory for storing all test artifacts.
+    The directory is created at session start and shared by all fixtures.
+
+    Directory naming includes BUILD_NUMBER if running in Jenkins CI environment.
+    """
+    timestamp = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+
+    # BUILD_NUMBER and WORKSPACE are specific to Jenkins
+    build_number = os.environ.get('BUILD_NUMBER')
+    if build_number:
+        logging.debug(f"Use Jenkins CI build number: {build_number}")
+        workspace = os.environ.get('WORKSPACE')
+        log_dir = f"{workspace}/report_{build_number}"
+    else:
+        log_dir = f"pytest-logs/report_{timestamp}"
+
+    # postpone directory creation only if some test failed
+    return log_dir
+
+
+# Record test timestamps to easily extract logs
+@pytest.fixture(scope='class', autouse=True)
+def bugreport_timestamp(request, host):
```

Comment:
"Record test start and end timestamps"?  Could make sense to be a docstring?

The "to easily extract logs" could have more details, as is it is no obvious how those timestamps help that.